packages/dataviews/src/components/dataviews/index.tsx

```diff
@@ -194,6 +194,21 @@ function DataViews< Item >( {
 		}
 	}, [ hasPrimaryOrLockedFilters, isShowingFilter ] );
 
+	// Show filter panel when filters are added (but allow manual hiding).
+	const [ prevFiltersLength, setPrevFiltersLength ] = useState(
```

Comment:
My instinct is that using a ref would be safer and more performant, seeing that `setPrevFiltersLength` won't be used in a different context. We can avoid rerenders and having `prevFiltersLength` in the dependency array, and then updating its value opens the door for infinite loops.